### PR TITLE
Wrapp all errors

### PR DESCRIPTION
**Pull request by guggero**
In this PR we wrap all errors so we can properly use `errors.Is()` to find the cause for any error. This is especially useful for database related problems, so the underlying database error can be unwrapped.

### CODE REVIEW

File: waddrmgr/error.go [hunk 1]
```diff
@@ -207,6 +207,11 @@ func (e ManagerError) Error() string {
 	return e.Description
 }
 
+// Unwrap returns the underlying error, if any.
+func (e ManagerError) Unwrap() error {
```
Comment (Roasbeef) [inline waddrmgr/error.go:211]: 👍

File: wtxmgr/tx.go [hunk 2]
```diff
@@ -834,7 +834,7 @@ func (s *Store) UnspentOutputs(ns walletdb.ReadBucket) ([]Credit, error) {
 		rec, err := fetchTxRecord(ns, &op.Hash, &block)
 		if err != nil {
 			return fmt.Errorf("unable to retrieve transaction %v: "+
-				"%v", op.Hash, err)
+				"%", op.Hash, err)
```
Comment (Roasbeef) [inline wtxmgr/tx.go:837]: Should be `%w`. 

We also need to tag this module after merge to then update the main module.

**Review by Roasbeef — CHANGES_REQUESTED** [review]

File: wtxmgr/tx.go (hunk 2 above)
Comment (guggero) [reply wtxmgr/tx.go]: Oops, nice catch. Fixed.

**Comment by guggero** [comment]
Did some more regex foo and found a few more instances. Now there should really be none left (that aren't in test code).

**Review by Roasbeef — APPROVED** [review]
LGTM 🧩

**Pull request merged**